### PR TITLE
Falco initial GEP proposal

### PR DESCRIPTION
**Pull request by marwinski**
**How to categorize this PR?**

/area TODO
/kind TODO

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

File: docs/proposals/27-falco-extension.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+- [References](#references)
+
+## Summary
+
+Falco is a cloud native runtime security tool for Linux operating systems. It
+is designed to detect and alert on abnormal behavior and potential security 
+threats in real-time [1]. We have deployed it in the Gardener landscape and
+can confirm that it is well suited for detecting events that are not ordinary
+runtime events. This includes but is not limited to debugging activities,
+changing host configurations, or installing software in containers. Apart from
```
Comment (lsroe) [inline docs/proposals/27-falco-extension.md]:
```suggestion
changing host configurations, or installing software in containers. While these 
events may signify benign debugging activities they are most likely indicating 
malicious behavior.
```

File: docs/proposals/27-falco-extension.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+legal debugging activities those types of events most likely indicate malicious
+behavior.
+
+Falco is a powerful runtime threat detection engine but comes with limited
+means to deploy on many Kubernetes clusters or sophisticated event analysis
+and visualization for many clusters. These features are usually provided by
+commercial products with may themselves rely on Falco as the detection
+engine.
+
```
Comment (lsroe) [inline docs/proposals/27-falco-extension.md:47]:
```suggestion
Falco is a powerful runtime threat detection engine but does not in itself provide
means to be deploy in Kubernetes cluster environment.  It neither provides sophisticated 
event analysis and visualization capabilities for the monitoring of multiple Kubernetes cluster 
environments. These features are usually provided by commercial solutions which utilize the 
Falco detection engine. 
```

File: docs/proposals/27-falco-extension.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+This enhancement proposal describes a consistent way for deploying Falco into
+the Gardener infrastructure as well as into shoot clusters operated by Gardener
+users in a "Gardener native" way. This includes managing it, providing Falco
+updates and bringing a central infrastucture for viewing and analyzing events.
+
+## Motivation
+
+While we have invested a lot into Gardener security and carry out regular
+penetration tests we don't have a lot of runtime insights into our Kubernetes
```
Comment (lsroe) [inline docs/proposals/27-falco-extension.md]:
```suggestion
penetration tests we currently do not have the ability to precisely monitor runtime behaviours within our Kubernetes
```

File: docs/proposals/27-falco-extension.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+clusters unless we install 3rd party commercial tools. This includes but is
+not limited to security breaches through misconfigurations or 0day
```
Comment (lsroe) [inline docs/proposals/27-falco-extension.md]:
```suggestion
not limited to security breaches through misconfigurations or 0-day 
```

File: docs/proposals/27-falco-extension.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+### Non-Goals
+
+We do not compete with commercial products. While we have plans to develop
+Falco plugins, we will not provide features such as image-, or malware scanning
+that is provided by commercial vendors.
+
+## Proposal
+
+We are planning to provide a Gardener extension that deploys Falco and 
+Falcosidekick into shoot clusters. All events that are generated by Falco are
```
Comment (lsroe) [inline docs/proposals/27-falco-extension.md:93]: maybe a super short description if what the sidekick actually is

, Falco's own event forwarding daemon,

File: docs/proposals/27-falco-extension.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+```
+  resources:
+  - name: customRules
+    resourceRef:
+      apiVersion: v1
+      kind: ConfigMap
+      name: my-custom-rules
+```
+
+Events that are generated by Falco are forwarded to Falcosidekick that is
```
Comment (lsroe) [inline docs/proposals/27-falco-extension.md:203]:
```suggestion
Events that are generated by Falco are forwarded to Falcosidekick which is 
```

File: docs/proposals/27-falco-extension.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+
+- event rate limiting based on cluster identities
+- strict validation of events
+
+The implementation of the event store is deliberately not specified in this
+enhancement proposal. There are however a number of requirements toward the
+event store:
+
+- the event store must be able to store events generated from a whole Gardener 
+  landscape with potentially thousands of clusters.
```
Comment (lsroe) [inline docs/proposals/27-falco-extension.md:267]:
```suggestion
  landscape with potentially thousands of clusters
```

**Review by lsroe — APPROVED** [review]

File: docs/proposals/27-falco-extension.md (hunk 2 above)
Comment (marwinski) [reply docs/proposals/27-falco-extension.md:47]: I would leave that in.

File: docs/proposals/27-falco-extension.md (hunk 3 above)
Comment (marwinski) [reply docs/proposals/27-falco-extension.md]: re-phrased in a different way

**Pull request merged**